### DISCUSSION TITLE
Configuration Python Windows

### DISCUSSION THREAD

**Discussion by Qunal** [smartcontractkit/full-blockchain-solidity-course-py#597] [Q&A] [answered]
I wanted some hints. I have a python 3.8 anaconda setup on Win 10 64bit . Now wanted to choose the best option to minimize issues.
Python 3.8 or 3.9 , VS, and WSL

Install WSL (windows subsystem for linux) seems to be least pain. Now to avoid conflicts with my other project and assuming anaconda environments may not be the best course 
Do I install VS Code, new python and addons and virtualenv will be outside WSL ( in windows not under WSL/Ubuntu) and there will be a simple way for the code to be tested from VS by executing within WSL

Some years ago I had used Docker and it was painful to debug .

Any general links on WSL/ Usage of VS and other tools

**Comment by cromewar** [comment, marked as answer]
Hello @Qunal deppends on each user preference but I always try to push people to use WSL for blockchain dev, here are some reasons:

1. Windows has a lot of troubles with web3 libs and requires you to install visual studio community and the python development tools (cython) not mentioning sometimes you'll need some extra Visual C++ Build tools.
2. Anaconda is great, for machine learning and Data science, but some users have reported they also have problem with web3 libraries.
3. WSL is easiest and you can run visual studio code from it `code .` and the extensions you install on it will be managed separately from the windows instance.

You can find a guide about how to setup your windows machine for blockchain dev on the course index "windows support appart". Finally, if you want to use visual studio instead of visual studio code you could refer to [this](https://docs.microsoft.com/en-us/answers/questions/72614/using-vs2019-with-wsl-windows-subsystem-for-linux.html) to make it work with WSL.

Best regards, Cromewar.

**Reply by Qunal** [reply]
Thanks. As I suspected. Will install WSL and use VS with remote extensions. So basically almost all the work will be under a Linux OS with VS sitting in windows. Will have some teething issues with Nano editors etc.

**Reply by Qunal** [reply]
Fresh question. When do we activate a virtual environment? I assume we need to give it a name and make it default before installing brownie etc? 

My versions 
ubuntu 20.04.3
node 17.3.0
python 3.8.10 
pip 21.3.1

**Reply by cromewar** [reply]
You use the virtualenvs when you want it, let's se an example.
1. Create the virtual env: `virtualenv NameYouWant`
2. Activate it: `source NameYourWant/bin/activate`
3. install the packages you need: `pip install eth-brownie`.
4. When finish working, deactivate it: `deactivate`

Note, remember to have to be on the env directory to activate it, for example if you have a env on your Documents folder and you try to activate it from home folder you should provide all the path on which the env is, so is easiest  to first be sure to be on the en directory.

**Reply by Qunal** [reply]
Thanks for a prompt reply over Christmas.
